rust/src/dhcp/dhcp.rs

```diff
@@ -17,12 +17,13 @@
 
 use crate::applayer::{self, *};
 use crate::core;
-use crate::core::{ALPROTO_UNKNOWN, AppProto, Flow, IPPROTO_UDP};
+use crate::core::{ALPROTO_UNKNOWN, Flow, IPPROTO_UDP};
 use crate::dhcp::parser::*;
+use crate::sys::AppProtoEnum;
 use std;
 use std::ffi::CString;
 
-pub(super) static mut ALPROTO_DHCP: AppProto = ALPROTO_UNKNOWN;
+pub(super) static ALPROTO_DHCP: AppProto = AppProtoEnum::ALPROTO_DHCP as AppProto;
```

Comment:
Can do this for all protocols.  The cast is because the enum is naturally an i32, however we use an i16 over on C to save space.